config/v1/types_feature.go

```diff
@@ -131,11 +131,40 @@ type FeatureGateAttributes struct {
 	// +required
 	Name FeatureGateName `json:"name"`
 
+	// requiredMinimumComponentVersion is a list of component/version pairs that declares the is the lowest version the given
+	// component may be in this cluster.
+	// Currently, the only supported component is Kubelet, and setting a required minimum kubelet component will set the
+	// minimumKubeletVersion field in the nodes.config.openshift.io CRD.
+	// +kubebuilder:validation:MaxItems:=1
+	// +listType=map
+	// +listMapKey=component
+	// +openshift:enable:FeatureGate=MinimumKubeletVersion
+	// +optional
+	RequiredMinimumComponentVersions []RequiredMinimumComponentVersion `json:"requiredMinimumComponentVersions,omitempty"`
+
 	// possible (probable?) future additions include
 	// 1. support level (Stable, ServiceDeliveryOnly, TechPreview, DevPreview)
 	// 2. description
 }
 
+// RequiredMinimumComponentVersion is a pair of Component and Version that specifies the required minimum Version of the given Component
+// to enable this feature.
+type RequiredMinimumComponentVersion struct {
+	// component is the entity whose version must be above a certain version.
+	// +required
+	Component RequiredMinimumComponent `json:"component"`
```

Comment:
```suggestion
	// component is the entity whose version must be above a certain version.
	// The only valid value is Kubelet.
	// When set to Kubelet, the minimumKubeletVersion field in the nodes.config.openshift.io resources will be set to ???
	// +required
	Component RequiredMinimumComponent `json:"component"`
```
What happens if I have multiple gates and one declares 1.29 and one declares 1.30? Is it the highest value that wins here?

Reply:
neither "win" if the min version is 1.29 the 1.30 one will be filtered but 1.29 will remain. if min version is 1.28 or unset then both will be filtered

Reply:
Where does the min version get compared from?

I'm guessing something is watching nodes, and sets the `nodes.config` min version to whatever the minimum is, then the feature gate controller observes changes to `nodes.config` and updates the enabled/disabled accordingly?

So, would it be true that I have to move all nodes in my cluster to 1.30 for a 1.30 gate to be enabled, having just a single 1.29 node would force the feature off?

Do we have ratcheting on this? Can the cluster go backwards?

Reply:
>I'm guessing something is watching nodes, and sets the nodes.config min version to whatever the minimum is, then the feature gate controller observes changes to nodes.config and updates the enabled/disabled accordingly?

yeah I have to figure out the watch mechanism but that's my goal

> So, would it be true that I have to move all nodes in my cluster to 1.30 for a 1.30 gate to be enabled, having just a single 1.29 node would force the feature off?

technically all you'd have to do is set the nodes.config minimumKubeletVersion field to 1.30, but the admission plugin for minimumKubeletVersion blocks if any nodes are currently in the cluster lower than that version, so yeah that feature would be forcably disabled 

> Do we have ratcheting on this? Can the cluster go backwards?
I see no reason why it couldn't go backwards, but features may get turned off if so

Reply:
so you'd prefer we set it so minimum kubelet version (once set) always increases?

Reply:
We don't support Y stream downgrades last I checked, so I'm not sure why we'd support Y stream downgrades of the minimum kubelet, might be worth checking with others to see if they agree on this though

Reply:
for posterity: it was determined we need downgrades